wp-cli/vip-option-autoload.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Manage autoloaded options
+ */
+class Option_Autoload extends WP_CLI_Command {
+
+	/**
+	 * Set autoload for option
+	 *
+	 * ## OPTIONS
+	 *
+	 * <option>
+	 * : Name of option
+	 *
+	 * <yn>
+	 * : yes or no
+	 * ---
+	 * options:
+	 *   - yes
+	 *   - no
+	 * ---
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Set autoload value to 'no'for 'home'
+	 *     $ wp option autoload set home no
+	 *     Success: Autoload changed. Cache flushed.
+	 *
+	 */
+	function set( $args, $assoc_args ) {
+
+		list( $option, $yn ) = $args;
+
+		$yn = $this->validate_yn( $yn );
+
+		wp_protect_special_option( $option );
+
+		global $wpdb;
+		$option_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload from {$wpdb->options} where option_name = %s", $option ) );
+
+		if ( is_null( $option_autoload ) ) {
+			WP_CLI::error( "Option does not exist." );
+		}
+
+		if ( $option_autoload === $yn ) {
+			WP_CLI::error( sprintf( "Option autoload already set to '%s'.", $yn ) );
+		}
+
+		$wpdb->get_results( $wpdb->prepare( "UPDATE {$wpdb->options} SET autoload = %s where option_name = %s", $yn, $option ) );
+
+		$check_option = $wpdb->get_var( $wpdb->prepare( "SELECT autoload from {$wpdb->options} where option_name = %s", $option ) );
+
+		if ( $check_option === $option_autoload ) {
+			WP_CLI::error( "Option not updated." );
+		}
+
+		$alloptions_before = wp_cache_get( 'alloptions', 'options' );
+		wp_cache_delete( 'alloptions', 'options' );
+		wp_load_alloptions(); // populate cache
+		$alloptions_after = wp_cache_get( 'alloptions', 'options' );
+
+		$cache_success = false;
+		switch ( $check_option ) {
+			case 'no' :
+				$cache_success = ( isset( $alloptions_before[ $option ] ) ) && ( ! isset( $alloptions_after[ $option ] ) );
+			break;
+			case 'yes' :
+				$cache_success = ( ! isset( $alloptions_before[ $option ] ) ) && ( isset( $alloptions_after[ $option ] ) );
+			break;
+		}
+
+		WP_CLI::success( sprintf(
+			'Autoload changed.%s',
+			WP_CLI::colorize( $cache_success ? ' Cache flushed.' : ' %rCache flush failed.%n' )
```

Comment:
The spacing looks a little odd here.

Maybe something like this is better?
```suggestion
			'Autoload changed. %s',
			WP_CLI::colorize( $cache_success ? 'Cache flushed.' : '%rCache flush failed.%n' )
```

Reply:
It's the difference between:

`Success: Autoload changed. Cache flushed.`

and

`Success: Autoload changed.Cache flushed.`

The space should stay for formatting purposes.

Reply:
It's needed for formatting. Otherwise there's no space between the cache statement and the rest of the output.

Reply:
Yes but when using something like `sprintf()`, you would usually put the spaces in the format string instead of the parameters being inserted. That way, the formatting is handled by the formatting string alone and is not a concern when inserting the parameters. Still, it's minor matter.

Reply:
Whoops, sorry I'm blind 🤦‍♂